web/src/pages/api/public/dataset-run-items.ts

```diff
@@ -93,6 +105,41 @@ export default withMiddlewares({
         },
       });
 
+      /********************
+       * ASYNC RUN ITEM EVAL *
+       ********************/
+
+      if (redis && env.NEXT_PUBLIC_LANGFUSE_CLOUD_REGION) {
+        const queue = DatasetRunItemUpsertQueue.getInstance();
+        if (queue) {
+          await queue.add(
+            QueueJobs.DatasetRunItemUpsert,
+            {
+              payload: {
+                projectId: auth.scope.projectId,
+                type: "dataset" as const,
+                datasetItemId: datasetItemId,
+                traceId: finalTraceId,
+                observationId: observationId ?? undefined,
+              },
+              id: randomUUID(),
+              timestamp: new Date(),
+              name: QueueJobs.DatasetRunItemUpsert as const,
+            },
+            {
+              attempts: 3, // retry 3 times
+              backoff: {
+                type: "exponential",
+                delay: 1000,
+              },
+              delay: 180_000, // 180 seconds
```

Comment:
The schema of the `jobExecution` has a fk relationship enforced on `traceId`. I added a generous delay to ensure we do not attempt to create a dataset eval job execution before the trace has been created via the ingestion queue. Do you have any insight here how to make this approach more robust? Once added to the queue, we validate the existence of the trace, and retry if it's not found.